flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaOptions.java

```diff
@@ -698,6 +706,49 @@ private static boolean hasKafkaClientProperties(Map<String, String> tableOptions
         throw new TableException("Unknown value fields strategy:" + strategy);
     }
 
+    public static ReadableConfig completeAvroConfluentSubject(ReadableConfig tableOptions) {
+        Configuration configuration = (Configuration) tableOptions;
+        final String format = configuration.getString(FORMAT);
+        final String valueFormat = configuration.getString(VALUE_FORMAT);
+        final String avro = "avro-confluent";
+        final String debezium = "debezium-avro-confluent";
+        if (avro.equals(format)) {
+            completeKeyAvroConfulentSubject(configuration, avro, false);
+        } else if (debezium.equals(format)) {
+            completeKeyAvroConfulentSubject(configuration, debezium, false);
+        } else if (avro.equals(valueFormat)) {
+            completeKeyAvroConfulentSubject(configuration, avro, true);
+        } else if (debezium.equals(valueFormat)) {
+            completeKeyAvroConfulentSubject(configuration, debezium, true);
+        }
+        return configuration;
+    }
+
+    private static void completeKeyAvroConfulentSubject(
```

Comment:
We can make this method more generic, e.g. `autocompleteSchemaRegistrySubject(Configuration conf, String format)` and consider `format=?`, `key.format=?`, `value.format=?` in it.